### PR TITLE
Added typescript type definition file as well as minor documentation changes.

### PR DESCRIPTION
**Pull request by schalk-b**

### CODE REVIEW

File: src/chain.js [hunk 1]
```diff
@@ -61,10 +61,10 @@ class Chain {
  * @param {Chain} chain - Chain to compose the commit of.
  * @param {string} ecAddress - Entry Credit address that pays for the commit, either private (Es) or public (EC). 
  * If a public EC address is provided it is necessary to provide the signature of the commit as a 3rd argument (use case for hardware wallets)
- * @param {string|Buffer} [sig] - Optional signature of the commit (composeChainLedger). Only necessary if a public EC address was passed as 2nd argument.
+ * @param {string|Buffer} [signature] - Optional signature of the commit (composeChainLedger). Only necessary if a public EC address was passed as 2nd argument.
  * @returns {Buffer} - Chain commit.
  */
-function composeChainCommit(chain, ecAddress, sig) {
+function composeChainCommit(chain, ecAddress, signature) {
```
Comment (PaulBernier) [inline src/chain.js]: I agree that now that this argument is publicly documented it is better to have a full name for the variable, but only renaming the argument is breaking the implementation inside the function. Actually inside the function I was also declaring a variable named `signature`, one possibility would be to rename the old `signature` to `sig` and vice versa.

**Comment by PaulBernier** [comment]
Thanks a lot for that PR!

I'm not familiar with Typescipt, is adding `"types": "factom.d.ts"` to package.json enough for all Typescript devs to benefit from those definitions out of the box? Should there be any addition instructions in the README for usage or we can legitimately expect that any Typescript developer will know how to leverage your type definition file?

File: src/chain.js (hunk 1 above)
Comment (schalk-b) [reply src/chain.js]: Good catch! How did I miss that. Thanks for reviewing.

**Comment by schalk-b** [comment]
Yup, just referencing types in package.json does the trick. Once they have typescript setup on their project, typing for the library will work straight after doing an npm install factom.

But I do think we should add information about typescript support in the README, I'll also do up a minimal starter project as an example.

I won't be able to get to this until next weekend.

**Comment by PaulBernier** [comment]
I just added 2 small util functions `generateRandomFctAddress` and `generateRandomEcAddress`: https://github.com/PaulBernier/factomjs/blob/master/src/addresses.js#L224

If you get a chance to add those also to your PR. Otherwise it'll be a chance for me to try myself to add those to the typescript file. Thanks!

**Comment by PaulBernier** [comment]
Hey @schalk-b will you be able to make those changes soon? Otherwise I can just merge the PR as is and I will make the changes.

**Pull request merged**